source/azure_iot_hub_client.c

```diff
@@ -815,7 +847,7 @@ AzureIoTHubClientResult_t AzureIoTHubClient_SendTelemetry( AzureIoTHubClient_t *
     }
     else
     {
-        xMQTTPublishInfo.xQOS = eAzureIoTMQTTQoS1;
+        xMQTTPublishInfo.xQOS = xQOS;
```

Comment:
>xQOS [](http://example.com/codeflow?start=32&length=4)

xQOS is of different type then MqttQoS, should this result in build failure?

Reply:
I have it casted now and checked in unit tests to make sure the qos aligns. Whether it should have broken the build or not I think is worthwhile to look into the build flags

Reply:
how about 

xQOS == eAzureIoTHubMessageQoS1 ? eAzureIoTMQTTQoS1 : eAzureIoTMQTTQoS0;


even though we own both enum... its better not to do unnecessary casts